Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -1143,10 +1156,10 @@ WarpXParticleContainer::ApplyBoundaryConditions (ParticleBoundaries& boundary_co
 #endif
                                                               z, zmin, zmax,
                                                               ux[i], uy[i], uz[i], particle_lost,
-                                                              boundary_conditions);
+                                                              this->m_boundary_conditions, engine);
 
                     if (particle_lost) {
-                        p.id() = -1;
+                        p.id() = -1; //-p.id();
```

Comment:
This will need to be squared away before integration.

Reply:
Yeap, just copying here for reference. When setting the ids to `p.id() = -p.id();` there were failures with laser particles in the CI tests.

Reply:
@atmyers Looks like the fix to skip particles that already have their ID's set to a negative number fixed the issue.

Reply:
Nice! I think this is ready to go in.